lib/file.js

```diff
@@ -169,17 +169,18 @@ class SourceFile extends File {
      * Instead, we generate this utility object for the runtime to use:
      * @example
      * ```js
-     * stringifyLambda({name: 'f', parameters: [['p','T']], returns: 'number'})  // { (p: T): number, __parameters: { p: T } }
+     * stringifyLambda({name: 'f', parameters: [['p','','T']], returns: 'number'})  // { (p: T): number, __parameters: { p: T } }
      * ```
      */
     static stringifyLambda({name, parameters=[], returns='any', kind, initialiser, isStatic=false, callStyles={positional:true, named:true}}) {
-        const parameterTypes = parameters.map(([n, t]) => `${normalise(n)}: ${t}`).join(', ')
+        const parameterTypes = parameters.map(([n, mod='', t]) => `${normalise(n)}${mod}: ${t}`).join(', ')
         const parameterTypeAsObject = parameterTypes.length
             ? createObjectOf(parameterTypes)
             : empty
         const callableSignatures = []
         if (callStyles.positional) {
-            callableSignatures.push(`// positional\n(${parameterTypes}): ${returns}`)
+            const paramTypesPositional = parameters.map(([n, , t]) => `${normalise(n)}: ${t}`).join(', ') // must not include ? modifiers
```

Comment:
depending on how badly we want optional positional parameters:
```suggestion
            // the first optional parameter makes all succeeding params optional as well
            let optional = parameters.at(0)?.[1] === '?'
            const paramTypesPositional = parameters.map(([n, mod, t]) => {
                optional ||= mod === '?'
                return `${normalise(n)}${optional ? '?' : ''}: ${t}`
            }).join(', ')
```

Reply:
I don't think we need this dynamic behavior just to avoid one parameter.
The current approach of offering the optionals in one call style only is more straight forward.